worker.py

```diff
@@ -0,0 +1,140 @@
+from email.mime import text
+import logging
+import pika
+import smtplib
+from sqlalchemy.orm import exc
+
+from mailbbgun import app
+from mailbbgun import models
+
+
+if app.config['DEBUG']:
+    logging.basicConfig(level=logging.DEBUG)
+
+
+_BANNER = """
+  __  __       _ _   ____  ____     _____
+ |  \/  |     (_) | |  _ \|  _ \   / ____|
+ | \  / | __ _ _| | | |_) | |_) | | |  __ _   _ _ __
+ | |\/| |/ _` | | | |  _ <|  _ <  | | |_ | | | | '_ \\
+ | |  | | (_| | | | | |_) | |_) | | |__| | |_| | | | |
+ |_|  |_|\__,_|_|_| |____/|____/   \_____|\__,_|_| |_|
+
+        _  ____________.---------.
+        \`'  __________|_________| o o o
+        /   (_)__]
+       |    |
+      .'   .'
+      |____]
+"""
+_DELIVERY_MODE_PERSISTENT = 2
+_LOG = logging.getLogger(__name__)
+
+
+class MailBBGunWorker():
+
+    def configure_rabbitmq(self):
+        self.connection = pika.BlockingConnection(
+            pika.ConnectionParameters(app.config['RABBITMQ_HOST'])
+        )
+        self.channel = self.connection.channel()
+        self.channel.queue_declare(queue='mailbbgun', durable=True)
+        self.channel.queue_bind(exchange='amq.direct', queue='mailbbgun')
+
+        self.retry_channel = self.connection.channel()
+        self.retry_channel.queue_declare(
+            queue='retry_delay',
+            durable=True,
+            arguments={
+                'x-message-ttl': app.config['RETRY_DELAY_MS'],
+                'x-dead-letter-exchange': 'amq.direct',
+                'x-dead-letter-routing-key': 'mailbbgun'
+            }
+        )
+
+    def process_message(self, ch, method, properties, body):
+        message_id = body.decode('UTF-8')
+        _LOG.info("Processing message id {}".format(message_id))
+        try:
+            message = self._get_message_by_id(message_id)
+            email = text.MIMEText(message.text)
+            email['Subject'] = message.subject
+            email['From'] = app.config['FROM_EMAIL']
+            email['To'] = message.to
+            self._send_email(email)
+            self._update_message_status(message, models.Status.delivered)
```

Comment:
Unfortunately the system would end up resending the email.  As I mentioned above, Rabbit is waiting for an ack before marking it as done.  If the worker dies here, rabbit will resend the message to another worker, and that worker will resend the email.

One possible way to mitigate this would be to add a new `PROCESSING` state to the message status.  We would at least know that something bad is happening if a worker starts processing a message and notices that the message is already in the PROCESSING state, although it would be tricky to figure out whether the email was sent or not.